design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -257,6 +258,50 @@ VMI status and in one case to pass the data (processed) to the
 
 - DownwardAPI
 
+##### Device Info
+
+In some cases the multus status annotation
+contains [device-info](https://github.com/k8snetworkplumbingwg/device-info-spec/blob/main/SPEC.md).
+This info may be valuable for the binding plugin sidecar. For example
+`SRIOV` and `VDPA` plugins need this info.
+
+As the multus status annotation is not there yet on `virt-launcher` pod
+creation, the suggested solution is to use downwardAPI volume and mount
+it to the sidecar.
+The downwardAPI volume will track a new `virt-launcher` pod
+annotation `kubevirt.io/network-info` that will contain the `network-info`
+of the networks that have `downwardAPI`.
+Very similar to the
+existing [SRIOV downwardAPI](https://github.com/kubevirt/kubevirt/pull/8226)
+solution which is mapping the networkName to the `pci-address` of the
+device and mounting the downwardAPI volume to the `virt-launcher` pod.
+
+**Note**: Multus provides device-info information since version v3.7.
+If an older version is used, the downward API is not supported,
+the annotation will have empty values.
+
```

Comment:
I think it will be nice defining the data model of the introduced annotation value, including what info it provides (PCI address..) in which format (map, json..)

Also when this annotation and downward API volume are be added (for every VM or only for those who require plugin sidecar)

Reply:
> I think it will be nice defining the data model of the introduced annotation value, including what info it provides (PCI address..) in which format (map, json..)

The vDPA example has it.

> Also when this annotation and downward API volume are be added (for every VM or only for those who require plugin sidecar)

It is written - `The downwardAPI volume will track a new `virt-launcher` pod
annotation `kubevirt.io/network-info` that will contain the `network-info`
of the networks that have `downwardAPI`.`
I find the exact details of when the volume is added are implementation details and shouldn't be part of a high level design.

Reply:
>The vDPA example has it.

Right but it is just an example I think it should be well defined.
Anyhow I am not insisting on it the proposal looks good.

> I find the exact details of when the volume is added are implementation details and shouldn't be part of a high level design.

At first I thought it could become backward-comparability concern, but it shouldn't be an issue:
On Kubevirt upgrade, for existing pods, virt-controller will just add the new annotation / new annotation value.
The annotation value wont propagate to the launcher app until the VM is migrated, 
which is fine since it cause no disruption.
New pods will be created with new annotation  volume.

Reply:
It is an API, so yes, a change should be considered as an API change.